### ISSUE TITLE
How well does pixi.js scale?

### ISSUE THREAD

**Issue by MattMcFarland** [GoodBoyDigital/pixi.js#1341]
Hi all,

I'm a long time gamedev enthusiast - currently working as a software engineer specializing in JS.  I have a ton of experience with JS and reading the tutorials made sense and it seems easy to pickup. 

While I've made a few games (and a little cash) off of the flash market several years ago, I've been thinking about diving into HTML 5.  pixi's cross-browser support seems very promising.  I'm just thinking of getting into it as sort of a hobby, and maybe meeting some cool folks and collaborate down the road.

However, I'm still not 100%, there's a lot of html 5 engines out there, and I'm sure each one is unique.  So please humor me as I describe my ideal game engine, and tell me if pixi is right (or not for me)

## TLDR? 
 Currently I'm looking for an html 5 engine that is:

* Lightweight

* Plays well with others (other JS frameworks such as underscore, and requireJS)

* Can scale. (extremely important) 

* Works well with CommonJS / AMD (requireJS, nodeJS, browserify; requireJS preferred)

Given the points above, how well does pixi.js fit?  Are there any demos out there that show the directory scaffold of a well engineered pixi.js?

**Comment by englercj** [comment]
> Lightweight

Wait until v3.

> Plays well with others (other JS frameworks such as underscore, and requireJS)

We export UMD so it should work with all module systems.

> Can scale. (extremely important)

Not sure what this even means. Do you mean screen size, project size, texture size, GPU ops, backwards compat...

> Works well with CommonJS / AMD (requireJS, nodeJS, browserify; requireJS preferred)

We export with a UMD wrapper. It does not run on nodejs yet, wait for v3.

A "well engineered pixi.js" project looks like what works best for your use case. Pixi is meant to be a general rendering engine for games, sites, charts, anything. Each project looks different, pixi is just one of many libraries your project will depend on.

**Comment by MattMcFarland** [comment]
Thanks for the reply.

As for "can scale" - sorry to be vague - I mean in strictly in maintainability.  as in scale well in volume of assets (separate script files, maintainable architecture, etc) I'll wait for v3, might play around with the code a bit to get my feet wet.

**Comment by englercj** [comment]
@MattMcFarland The architecture's maintainability is the primary goal of the v3 rewrite due to challenges we are facing in the maintainability of the core lib now. So hopefully that makes my answer "yes" when that work is done :)

**Comment by MattMcFarland** [comment]
:+1: Awesome! Thank you, and cant wait to see what happens.  IS there a forum or tracker discussing the goings on of v3? I'd love to watch and provide input if possible, even help if I can.  I'm an avid JS engineer with passion for good architecture :)  Or I can silently read at least...

**Comment by englercj** [comment]
Most conversation is happening on that thread. I'm doing most of the work for now and that is happening on my [dev-rewrite](https://github.com/englercj/pixi.js/tree/dev-rewrite) branch.

If you want to help out just hit send PRs to that branch. If you see something in the code that doesn't make sense, let me know. Feel free to run with any of the Phase4+ phases that seem interesting to you!

**Comment by englercj** [comment]
Obviously we also need examples and tests for the new code, so that is always a huge help as well!

**Comment by MattMcFarland** [comment]
Reading the proposal, and looks good so far. glad to see a lot of progress
already made.  I am especially pleased to see the decoupling a priority.
I'll look through more.  thanks again.

**Comment by englercj** [comment]
Anytime!

**Comment by MattMcFarland** [comment]
Just posted my thoughts here:
https://github.com/GoodBoyDigital/pixi.js/issues/1296#issuecomment-69439842
- please have a read I hope they help with phase 3.
